### ISSUE TITLE
RPC 0.5.29 on AMO

### ISSUE THREAD

**Issue by DJ-Leith** [RequestPolicyContinued/requestpolicy#664]
https://addons.mozilla.org/en-US/firefox/addon/requestpolicy-continued/versions/

Two are listed at AMO (on 2015-05-28):

> Version 0.5.29 Released May 27, 2015 169.0 kB Works with Firefox 4.0 and later, SeaMonkey 2.1 and later

When 'hovering' over the grey button "+ Add to Firefox",
I see a message: "Not available for your platform".

> Version 0.5.29b4 Released May 21, 2015 163.1 kB Works with Firefox 4.0 and later, SeaMonkey 2.1 and later

The button to "+ Add to Firefox" for Version 0.5.29b4 has 'yellow Warning stripes'
(expected if not Reviewed by Mozilla).

Version 0.5.29b4 works well (I've used it to convert many Profiles that had RP 0.5.28 to RPC,
and I've then updated those profiles to use RPC 1.0.beta9.3 - see #643).

As I saw this using a Profile that had RPC 1.0.beta9.3, I speculated that
'somehow AMO had detected I had RPC 1.0.xx' on my Firefox and it therefore showed
"Not available for your platform".

The other alternative is 'something is not OK'.

STR

Fresh Profile, no Extensions (Used Fx 38.0.1 on Windows).

Expected:
able to add Version 0.5.29 (possibly with Warning)

Actual:
> Version 0.5.29 Released May 27, 2015
Grey button "+ Add to Firefox",
I see a message: "Not available for your platform".

**Comment by myrdd** [comment]
Thanks for your report! It seems like I've only selected "linux" as platform. Now I've selected "all platforms". Can you please recheck?

**Comment by DJ-Leith** [comment]
Seems to be OK now.

When you click the "+ Add to Firefox", for Version 0.5.29, which now has the
'yellow Warning stripes', you then get a 2nd Warning:

>**Caution:** This add-on has not been reviewed by Mozilla and can't be installed on release versions of Firefox. Be careful when installing third-party software that might harm your computer.

Then you can install in Fx 38.0.1 using Windows.
So I could, despite the wording about "... release versions of Firefox ...", install it on Release.

I think this is 'expected behaviour' until Mozilla complete their Review.
I also think Mozilla will sign the 0.5.29 XPI.

After that, I expect 'Users will be able to find it AMO' and there will 'no longer be Warnings'.

**Comment by myrdd** [comment]
Yes, there's the warning because it's still not reviewed yet.

I have no idea why they say it *„can't be installed on release versions of Firefox“*. It's possible to install.

> I expect 'Users will be able to find it AMO' and there will 'no longer be Warnings'.

Correct. Currently its Queue Position is 189 of 213.

Can we close this issue @DJ-Leith?

**Comment by DJ-Leith** [comment]
> I have no idea why they say it "can't be installed on release versions of Firefox". It's possible to 
install.

Yes, that is why I documented that it can be installed (for other readers) and to confirm
to you - that 'all is OK on a Windows version of Firefox 38.0.1'.
It is possible that other readers here may see this (via links posted in other RPC issues).

I speculate wildly that Mozilla **might** use the same wording, at AMO, about 'unsingned Extensions'.
My understanding is that, when Mozilla enforce signing of all Extensions at AMO, it will **not** be 
possible to install unsigned Extensions on Release (but you will be able to install them on Nightly).


https://blog.mozilla.org/addons/2015/02/10/extension-signing-safer-experience/

https://blog.mozilla.org/addons/2015/04/15/the-case-for-extension-signing/

https://wiki.mozilla.org/Addons/Extension_Signing

> Can we close this issue @DJ-Leith?
Yes, I have now closed it.
